app/src/androidTest/java/com/mapbox/maps/testapp/featurestate/FeatureStateTest.kt

```diff
@@ -0,0 +1,108 @@
+package com.mapbox.maps.testapp.featurestate
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.mapbox.bindgen.Value
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.Point
+import com.mapbox.maps.Style
+import com.mapbox.maps.extension.style.layers.generated.circleLayer
+import com.mapbox.maps.extension.style.sources.generated.geoJsonSource
+import com.mapbox.maps.extension.style.style
+import com.mapbox.maps.testapp.BaseMapTest
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+/**
+ * Instrumented test for FeatureState api.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+@RunWith(AndroidJUnit4::class)
+class FeatureStateTest : BaseMapTest() {
+  private lateinit var countDownLatch: CountDownLatch
+
+  override fun loadMap() {
+    countDownLatch = CountDownLatch(1)
```

Comment:
seems we are reusing the same countDownLatch reference for both test and loadMap, shall we use separate latch instead?